### ISSUE TITLE
Crashes when app quickly loses then regains focus

### ISSUE THREAD

**Issue by damienzhang** [shoes/shoes-deprecated#142]
I'm experiencing a crash on the latest shoes 3 on Windows XP. If I click inside the shoes app, click outside, then click inside again quickly (within one or two seconds), I get a MSVC++ runtime error: "This application has requested the Runtime to terminate in an unusual way."

Is there some time-sensitive thread that's running when the window focus changes?

**Comment by steveklabnik** [comment]
There is some weirdness going on with threads in general. This works every
time? That'd be great... I'm gonna be working on windows this week.
On Sep 14, 2011 3:35 PM, "damienzhang" <
reply@reply.github.com>
wrote:

**Comment by damienzhang** [comment]
Yup, it's working every time for me. I tried it on a VM image of Windows 7 and it happens there as well.

**Comment by steveklabnik** [comment]
Awesome... sorta. At least I have a way to reproduce it. :)

**Comment by ccoupe** [comment]
I tried this on my XP VM and yes it does crash shoes but you have to be quick on the clicks/mouse moves. FWIW I only allocate one core of my Linux box to the XP VM.

**Comment by damienzhang** [comment]
Yeah you're right, I may have overestimated the delay it takes. It's more like 500ms or so on my machine. I might be able to investigate this further on the weekend, although it would be my first time looking at the shoes codebase :o)

**Comment by ccoupe** [comment]
I forgot to mention I was clicking on the background of the Shoes screen (no script loaded or running) so that would be Shoes keypress

I would be suspicious that mouse in/out/down event/resource isn't released or cleared or improperly reused if there is no Shoes+Script handler for the click. Then again that could be way off base. Good luck.

**Comment by eaffleck** [comment]
This is popping up in the hackety-hack questions as well, so I've done a little testing on Win XP SP3 to expand on this. Using HH downloaded last week.

Confirmed the above lost-focus-get-focus error with hackety-hack when clicking with mouse (get a couple of apps open and click between them, about 1-2 seconds between changes) and fails after a handful of app changes (3-6 reps).

Also used alt-tab to switch between apps and could not reproduce error. Tried faster switching and more of them (20 changes) and had no failures. Then click back on HH, and boom.

It _appears_ that error will turn up faster if I've tried to open Shoes console and then click on HH but not confident.

Finally, tried with a Shoes window opened by a running program in HH (so both windows are sourced from Shoes) and clicked (furiously - 20 times, really quick) between them without error. Returned to clicking between Shoes and another app and got error within 4 changes of app.

**Comment by ccoupe** [comment]
This doesn't seem to happen with Shoes 3.2 (windows gtk). Fixed?

**Comment by PragTob** [comment]
I'd mark it as fixed if we made 3.2 the official version in this repo and push the changes here, which we should totally talk about :-)

**Comment by ccoupe** [comment]
I'm certainly open to talking about 3.2.  It doesn't seem that my pull request will be answered soon and I'm fine with that if that's what the community wants. However, I see no reason to leave bug reports hanging out that have been fixed - there's a lot more Linux bugs I could have commented on as fixed but I started with the oldest.
